stl/inc/ranges

```diff
@@ -1328,8 +1328,190 @@ namespace ranges {
+                is_nothrow_copy_constructible_v<_Iterator>&& noexcept(--_Current)) /* strengthened */ {
+                auto _Tmp = *this;
+                --*this;
+                return _Tmp;
+            }
+
+            constexpr _Iterator& operator+=(difference_type _Off) noexcept(
+                noexcept(_Current += _Off)) /* strengthened */ {
+                _Current += _Off;
+                return *this;
+            }
+            constexpr _Iterator& operator-=(difference_type _Off) noexcept(
+                noexcept(_Current -= _Off)) /* strengthened */ {
+                _Current -= _Off;
+                return *this;
+            }
+            _NODISCARD constexpr const _Wx& operator[](difference_type _Idx) const noexcept {
+                return *(*this + _Idx);
+            }
+
+            _NODISCARD_FRIEND constexpr bool operator==(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(_Left._Current == _Right._Current)) /* strengthened */ {
+                return _Left._Current == _Right._Current;
+            }
+            _NODISCARD_FRIEND constexpr auto operator<=>(const _Iterator& _Left, const _Iterator& _Right) noexcept(
+                noexcept(_Left._Current <=> _Right._Current)) /* strengthened */ {
+                return _Left._Current <=> _Right._Current;
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator+(_Iterator _Iter, difference_type _Off) noexcept(
+                noexcept(_Iter += _Off)) /* strengthened */ {
+                return _Iter += _Off;
+            }
+            _NODISCARD_FRIEND constexpr _Iterator operator+(difference_type _Off, _Iterator _Iter) noexcept(
+                noexcept(_Iter += _Off)) /* strengthened */ {
+                return _Iter += _Off;
+            }
+
+            _NODISCARD_FRIEND constexpr _Iterator operator-(_Iterator _Iter, difference_type _Off) noexcept(
+                noexcept(_Iter -= _Off)) /* strengthened */ {
+                return _Iter -= _Off;
+            }
+            // clang-format off
+            _NODISCARD_FRIEND constexpr difference_type operator-(const _Iterator& _Left,const _Iterator& _Right)
+                noexcept(noexcept(_Left._Current - _Right._Current)) /* strengthened */ {
+                // clang-format on
+                return static_cast<difference_type>(_Left._Current) - static_cast<difference_type>(_Right._Current);
+            }
+        };
+
+        /* [[no_unique_address]] */ _Movable_box<_Wx> _Value{};
+        /* [[no_unique_address]] */ _Bo _Bound{};
+
+        template <class _Tuple1, class _Tuple2, size_t... _Indexes1, size_t... _Indexes2>
+        _NODISCARD_CTOR constexpr repeat_view(
+            _Tuple1& _Val1, _Tuple2& _Val2, index_sequence<_Indexes1...>, index_sequence<_Indexes2...>) noexcept
+            : _Value(in_place, _Tuple_get<_Indexes1>(_STD move(_Val1))...),
+              _Bound(_Tuple_get<_Indexes2>(_STD move(_Val2))...) {}

```

Comment:
This ctor can't be unconditionally `noexcept` - it's totally valid for these constructions to throw exceptions.

Reply:
I was hoping to get away with that since it's only used by the public one that is conditional; because I'm not exactly sure how to spell the nothrow requirement here.

Reply:
How about `is_nothrow_constructible_v<_Wx, tuple_element_t<_Indexes1, _Tuple1>...>`?